packages/date-adapters/src/types.ts

```diff
@@ -0,0 +1,382 @@
+/**
+ * Represents the date object of a date framework.
+ *
+ * If the DateFrameworkTypeMap is empty, it defaults to any; otherwise, it is a union of all types in the map.
+ */
+
+export type DateFrameworkType = keyof DateFrameworkTypeMap extends never
+  ? any
+  : DateFrameworkTypeMap[keyof DateFrameworkTypeMap];
+
+/**
+ * Enum representing possible date detail error types.
+ */
+export enum DateDetailErrorEnum {
+  /** Error type for unset values */
+  UNSET = "unset",
+  /** Error type for values that are not a date */
+  NOT_A_DATE = "not-a-date",
+  /** Error type for invalid date values */
+  INVALID_DATE = "date",
+  /** Error type for invalid month values */
+  INVALID_MONTH = "month",
+  /** Error type for invalid day values */
+  INVALID_DAY = "day",
+  /** Error type for invalid year values */
+  INVALID_YEAR = "year",
+}
+
+/**
+ * Represents an error detail for a date.
+ */
+export type DateDetailError = {
+  /** The error code */
+  type: DateDetailErrorEnum | string;
+  /** The error message */
+  message: string;
+};
+
+/**
+ * Provides a way to return date errors in a uniform way.
+ */
+export type DateDetail = {
+  /** The original entered value, if applicable */
+  value?: string;
+  /** The errors found by the parser */
+  errors?: DateDetailError[];
+};
+
+/**
+ * Represents the time components of a date.
+ */
+export type TimeFields = {
+  /** The hour component */
+  hour: number;
+  /** The minute component */
+  minute: number;
+  /** The second component */
+  second: number;
+  /** The millisecond component */
+  millisecond: number;
+};
+
+/**
+ * Options for configuring a date adapter.
+ *
+ * @template TLocale - The type of the locale.
+ * @template TInstance - The type of the date library instance.
+ */
+export type AdapterOptions<TLocale, TInstance> = {
+  /**
+   * The locale to be used by the date adapter.
+   */
+  locale?: TLocale;
+
+  /**
+   * The instance of the date library to be used by the adapter.
+   */
+  instance?: TInstance;
+};
+
+/**
+ * Recommended date formats for formatting date strings.
+ */
+export type RecommendedFormats =
+  | "dd"
+  | "dddd"
+  | "D"
+  | "DD"
+  | "DD MMM YYYY"
+  | "DD MMMM YYYY"
+  | "MMMM YYYY"
+  | "YYYY"
+  | string;
+
+/**
+ * Timezone options for date operations.
+ */
+export type Timezone = "default" | "system" | "UTC" | string;
+
+export type ParserResult<TDate extends DateFrameworkType> = {
+  date: TDate;
+} & DateDetail;
+
+/**
+ * Interface for a date adapter, providing methods for date manipulation and formatting.
+ *
+ * @template TDate - The type of the date object used by the adapter.
+ * @template TLocale - The type of the locale, defaulting to any.
+ */
+export interface SaltDateAdapter<
```

Comment:
Not 100% sure why certain function can accept optional `locale` as final arg, while the others doesn't.